libraries/AP_Scheduler/AP_Scheduler.cpp

```diff
@@ -389,6 +404,61 @@ void AP_Scheduler::Log_Write_Performance()
     AP::logger().WriteCriticalBlock(&pkt, sizeof(pkt));
 }
 
+// display task statistics as text buffer for @SYS/tasks.txt
+size_t AP_Scheduler::task_info(char *buf, size_t bufsize)
+{
+    size_t total = 0;
+
+    // a header to allow for machine parsers to determine format
+    int n = hal.util->snprintf(buf, bufsize, "TasksV1\n");
+
+    if (n <= 0) {
+        return 0;
+    }
+
+    // dynamically enable statistics collection
+    if (!(_options & uint8_t(Options::RECORD_TASK_INFO))) {
+        _options |= uint8_t(Options::RECORD_TASK_INFO);
```

Comment:
This should probably be .set() as then the GCS is informed of the parameter change.